src/azure-cli/azure/cli/command_modules/keyvault/_validators.py

```diff
@@ -189,6 +189,11 @@ def validate_principal(ns):
         raise argparse.ArgumentError(
             None, 'specify exactly one: --object-id, --spn, --upn')
 
+    attrs = ['object_id', 'spn', 'upn']
+    for attr in attrs:
+        if getattr(ns, attr):
+            setattr(ns, attr, getattr(ns, attr).strip('"'))
```

Comment:
for upn, is it safe to strip `"` ? what's the name convention? suggest only change object_id for now.

Reply:
> for upn, is it safe to strip `"` ? what's the name convention? suggest only change object_id for now.

Most of UPN names are email-like: https://docs.microsoft.com/en-us/azure/active-directory/hybrid/plan-connect-userprincipalname
As far as I know there is also a type of UPN for external users like this: `bim_microsoft.com#EXT#@AzureSDKTeam.onmicrosoft.com`

Both formats don't have leading or tailing quotes.
But in case that there are maybe some unknown formats may have quotes, I will keep the original format for UPN and SPN. Will investigate them again when we have new user ask.